silence/silence_test.go

```diff
@@ -364,41 +378,50 @@ func TestSilenceSet(t *testing.T) {
 	}
 	require.Equal(t, want, s.st, "unexpected state after silence creation")
 
-	// Overwrite silence 2 with new end time.
-	clock.Add(time.Minute)
-	start3 := s.nowUTC()
-
-	sil3 := cloneSilence(sil2)
-	sil3.EndsAt = start3.Add(100 * time.Minute)
-
+	// Can update silence without modifications.
+	sil3 := Clone(sil2)
 	require.NoError(t, s.Set(sil3))
 	require.Equal(t, sil2.Id, sil3.Id)
 
+	// Can update silence with new comment.
+	sil4 := Clone(sil3)
+	sil4.Comment = "c"
+	require.NoError(t, s.Set(sil4))
+	require.Equal(t, sil3.Id, sil4.Id)
+
+	// Extend sil4 to expire at a later time. This should not expire the
```

Comment:
These are existing test cases, but the variable names have been updated since I added two new cases above.